content/blog/2019-__-__-hydrology_task_view.md

```diff
@@ -0,0 +1,153 @@
+
+## State of Hydrology in R
+We will conduct a high-level assessment of the state of hydrology in R by visualizing network connectivity among package dependencies -- which packages use code from one package to accomplish a task. For context, we will evaluate our Hydrology task view against the **most** well-known collection of packages in R, the [tidyverse](https://www.tidyverse.org/), and the venerable [Environmetrics task view](https://CRAN.R-project.org/view=Environmetrics
+) which includes many packages for environmental analysis. To do that we need to write some R code and load our required packages:
+
+![](https://media.giphy.com/media/xT8qBmSnYDXS21ZvHO/giphy.gif)
+
+
+
+
+```r
+library(tidyverse)
+library(tidygraph)
+library(ggraph)
+library(ctv)
+library(tools)
+library(cranlogs)
+```
+
+
+
+
+We will use the handy `CRAN_package_db` function from the `tools` package (part of base R) which conveniently grabs information from the DESCRIPTION file of every package and turns it into a dataframe. Here we are only after a few columns so we will extract those right away:
+
+```r
+all_cran_packages <- CRAN_package_db()[,c("Package", "Imports")] 
+```
+
+We are also going to parse some of the info we receive from CRAN to make it a little easier to work with:
+
+```r
+tidied_cran_imports <- as_tibble(all_cran_packages) %>% 
+  separate_rows(Imports, sep = ",") %>% 
+  mutate(Imports = str_replace(Imports,"\\s*\\([^\\)]+\\)","")) %>%
+  mutate(Imports = str_trim(Imports, side = "both"))
+```
+
+### Package Connectivity
+
+First, let's take a look at the tidyverse. We can take the rare step of actually using a function from the tidyverse package (appropriately named `tidyverse_packages`), which unsurprisingly identifies those packages officially belonging to the tidyverse. We filter for those packages and their imports, convert to the `tbl_graph`, plot using `ggraph` to ultimately visualize package connectivity:
```

Comment:
Naive question. In "We can take the rare step..." is it rare because the tidyverse package is actually used to import a bunch of other packages in the tidyverse and functions from it are rarely called? I'm assuming you're not being facetious 🤔

Reply:
It is a little bit of a joke. Maybe a bit too "inside baseball" but you've got it exactly.

Reply:
I'm just learning R and am quite proud that I got this